shared/runtime/softtimer.c

```diff
@@ -30,11 +30,15 @@
 #include "py/runtime.h"
 #include "softtimer.h"
 
-#define TICKS_PERIOD 0x80000000
-#define TICKS_DIFF(t1, t0) ((int32_t)(((t1 - t0 + TICKS_PERIOD / 2) & (TICKS_PERIOD - 1)) - TICKS_PERIOD / 2))
-
 extern __IO uint32_t uwTick;
 
+STATIC inline int32_t ticks_diff(uint32_t t1, uint32_t t0) {
```

Comment:
please make it `static` because it should always be `static inline` regardless of how `STATIC` is defined

also, please move this function definition to just before `soft_timer_lt`, ie after all variables are declared/defined

Reply:
I made this change